documents/plan.md

```diff
@@ -1,11 +1,30 @@
 # Project plan
 
 ## Language evaluation
+I think what's really nice about my language is that I have a very clear MVP. I would consider my language a success if it could look at a given rhyme scheme for a poem and check that the rhyme scheme works. Perhaps the very next feature above the MVP would be suggesting alternative rhymes to words that do not fit the rhyme scheme. 
+
+As far as implementation practices go, I'm not totally sure. I hope my critiquers and reviewers have some sort of input on this. I've never relied on an API as much as I'm about to for this project, so I'm wondering if are some good practices that I should know about. Of course, I'll try to limit API calls as much as possible. 
+
+I think testing will be really important and I really want to make a good faith effort to follow test driven development practices for this project. I think the more rigorously I test, the more I will know that my code is doing what it should. Additionally, lots of testing will help me ensure that I know exactly what behaviours I want from my code. I'm a little nervous that I don't quite know what a real prototype for my program will look like, as opposed to an MVP; it seems there aren't too many steps in between. That said, I'll probably let testing of smaller functions and parts of the program help me granularize the problem and make sure everything along the way is going well.
```

Comment:
What do tests look like for your DSL? Do unit tests make sense? Which parts will you test?